styx-common/src/main/java/com/spotify/styx/state/RunState.java

```diff
@@ -172,8 +172,20 @@ public RunState created(WorkflowInstance workflowInstance, String executionId, S
     }
 
     @Override
+    public RunState dequeue(WorkflowInstance workflowInstance) {
+      switch (state()) {
+        case QUEUED:
+          return state(PREPARE);
+
+        default:
+          throw illegalTransition("dequeue");
+      }
+    }
+
+    @Override
     public RunState submit(WorkflowInstance workflowInstance, ExecutionDescription executionDescription) {
       switch (state()) {
+        case QUEUED: // for backwards compatibility
```

Comment:
Never mind! Modifying `triggerExecution` behaviour indeed means we need to allow for the `created` transition from QUEUED.